dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/sync/SingleEventDataSynchronizationService.java

```diff
@@ -0,0 +1,372 @@
+
+    return !progress.isSkipCurrentStage();
+  }
+
+  private void synchronizePageSafely(
+      int page, EventSynchronizationContext context, SystemSettings settings) {
+    try {
+      synchronizePage(page, context, settings);
+    } catch (Exception ex) {
+      log.error("Failed to synchronize page {}", page, ex);
+      throw new RuntimeException(
+          format("Page %d synchronization failed: %s", page, ex.getMessage()), ex);
+    }
+  }
+
+  private void synchronizePage(
+      int page, EventSynchronizationContext context, SystemSettings settings)
+      throws ForbiddenException, BadRequestException {
+    List<Event> events = fetchEventsForPage(page, context);
+
+    Map<Boolean, List<Event>> partitionedEvents = partitionEventsByDeletionStatus(events);
+    List<Event> deletedEvents = partitionedEvents.get(true);
+    List<Event> activeEvents = partitionedEvents.get(false);
+
+    syncEventsByDeletionStatus(activeEvents, deletedEvents, context, settings);
+  }
+
+  private List<Event> fetchEventsForPage(int page, EventSynchronizationContext context)
+      throws ForbiddenException, BadRequestException {
+
+    return eventService
+        .findEvents(
+            EventOperationParams.builder()
+                .programType(ProgramType.WITHOUT_REGISTRATION)
+                .program(context.getProgram())
+                .skipChangedBefore(context.getSkipChangedBefore())
+                .synchronizationQuery(true)
+                .includeDeleted(true)
+                .withSkipSyncDataElements(context.getSkipSyncDataElementsByProgramStage())
+                .build(),
+            PageParams.of(page, context.getPageSize(), false))
+        .getItems();
+  }
+
+  private Map<Boolean, List<Event>> partitionEventsByDeletionStatus(List<Event> events) {
+    return events.stream().collect(Collectors.partitioningBy(Event::isDeleted));
+  }
+
+  private void syncEventsByDeletionStatus(
+      List<Event> activeEvents,
+      List<Event> deletedEvents,
+      EventSynchronizationContext context,
+      SystemSettings settings) {
+
+    Date syncTime = context.getStartTime();
+    SystemInstance instance = context.getInstance();
+
+    if (!activeEvents.isEmpty()) {
+      List<org.hisp.dhis.webapi.controller.tracker.view.Event> activeEventDtos =
+          activeEvents.stream().map(EVENT_MAPPER::map).toList();
+      syncEvents(activeEventDtos, instance, settings, syncTime, false);
+    }
+
+    if (!deletedEvents.isEmpty()) {
+      List<org.hisp.dhis.webapi.controller.tracker.view.Event> deletedEventDtos =
+          deletedEvents.stream().map(this::toMinimalEvent).toList();
+      syncEvents(deletedEventDtos, instance, settings, syncTime, true);
+    }
+  }
+
+  private void syncEvents(
+      List<org.hisp.dhis.webapi.controller.tracker.view.Event> events,
+      SystemInstance instance,
+      SystemSettings settings,
+      Date syncTime,
+      boolean isDelete) {
```

Comment:
Instead of sending a boolean, directly send a parameter of type TrackerImportStrategy . 
And use that inside directly.